cockatrice/src/chatview.cpp

```diff
@@ -161,72 +161,115 @@ void ChatView::appendMessage(QString message, QString sender, UserLevelFlags use
     }
     cursor.setCharFormat(messageFormat);
     
-    int index = -1, bracketFirstIndex = -1, mentionFirstIndex = -1, urlFirstIndex = -1;
+    int index = -1, bracketFirstIndex = -1, mentionFirstIndex = -1, urlFirstIndex = -1, highlightWordFirstIndex = -1;
     bool mentionEnabled = settingsCache->getChatMention();
     const QRegExp urlStarter = QRegExp("https?://|\\bwww\\.");
     const QRegExp phraseEnder = QRegExp("\\s");
     const QRegExp notALetterOrNumber = QRegExp("[^a-zA-Z0-9]");
+    const QStringList highlightedWords = settingsCache->getHighlightWords();
     
     while (message.size())
     {
-        // search for the first [ or @
         bracketFirstIndex = message.indexOf('[');
         mentionFirstIndex = message.indexOf('@');
         urlFirstIndex = message.indexOf(urlStarter);
+        highlightWordFirstIndex = -1;
+
+        foreach (QString word, message.simplified().split(" "))
```

Comment:
I don't believe that will work. The foreach string splits the message up into an array, broken at each space. I'm open to looking for ways around the foreach loop, but this doesn't seem to be the way to go